pkg/controller/controller_utils.go

```diff
@@ -1140,7 +1135,15 @@ func PatchNodeTaints(ctx context.Context, c clientset.Interface, nodeName string
 		return fmt.Errorf("failed to marshal new node %#v for node %q: %v", newNodeClone, nodeName, err)
 	}
 
-	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
+	// Strip base diff node from RV to ensure that our Patch request will set RV to check for conflicts over .spec.taints.
```

Comment:
Can we specifically mention that this is needed because .spec.taints does not specify `patchMergeKey` and `patchStrategy` and adding them is no longer an option for compatibility reasons

Reply:
Done.

Also, added information about using other patch type and it's conflict with removal of a trait.